onnxruntime/core/framework/provider_bridge_ort.cc

```diff
@@ -378,9 +379,12 @@ struct ProviderHostImpl : ProviderHost {
 
   // Provider_DataTransferManager
   Status Provider_DataTransferManager__CopyTensor(const Provider_DataTransferManager* p, const Provider_Tensor& src, Provider_Tensor& dst, int exec_queue_id) override { return reinterpret_cast<const DataTransferManager*>(p)->CopyTensor(*reinterpret_cast<const Tensor*>(&src), *reinterpret_cast<Tensor*>(&dst), exec_queue_id); }
-
-  // Provider_IDataTransfer
-  void Provider_IDataTransfer__operator_delete(Provider_IDataTransfer* p) override { delete reinterpret_cast<Provider_IDataTransfer*>(p); }
+  const Provider_IDataTransfer* Provider_DataTransferManager__GetProviderDataTransfer(const Provider_DataTransferManager* p, const OrtDevice& src_device, const OrtDevice& dst_device) override {
+    auto* data_transfer = reinterpret_cast<const DataTransferManager*>(p)->GetDataTransfer(src_device, dst_device);
```

Comment:
>reinterpret_cast [](start = 26, length = 16)

Let's check these two casts if they are upcasting properly

Reply:
Yeah, I used to have Provider_IDataTransfer implemented by the ort side, now this is just an accessor function only for the DataTransferManager type, so that's why it's reversed from before. The reinterpret_cast is just the way to turn the Provider_* into the *.